Managers/UInputMan.cpp

```diff
@@ -939,172 +981,180 @@ namespace RTE {
-								}
-							}
-						}
+				std::array<InputElements, 4> elementsToCheck = {InputElements::INPUT_L_LEFT, InputElements::INPUT_L_UP, InputElements::INPUT_R_LEFT, InputElements::INPUT_R_UP};
+
+				for (std::size_t i = 0; i < elementsToCheck.size(); i += 2) {
+					int axisLeft{SDL_CONTROLLER_AXIS_INVALID};
+					int axisUp{SDL_CONTROLLER_AXIS_INVALID};
+					if (inputElements->at(elementsToCheck[i]).JoyDirMapped()) {
+						axisLeft = inputElements->at(elementsToCheck[i]).GetAxis();
+						aimValues.m_X = AnalogAxisValue(joystickIndex, axisLeft);
+						isAxisMapped |= axisLeft == axis;
 					}
-					aimValues.Reset();
-				}
-			}
-			for (int stick = 0; stick < joy[joystick].num_sticks; ++stick) {
-				for (int axis = 0; axis < joy[joystick].stick[stick].num_axis; ++axis) {
-					JOYSTICK_AXIS_INFO *joystickAxis = &joy[joystick].stick[stick].axis[axis];
-
-					if (joystickPlayer > Players::NoPlayer && deadZoneType == DeadZoneType::SQUARE && deadZone > 0.0F) {
-						// Adjust joystick values to eliminate values in deadzone. This heavily relies on AnalogAxisValue method of processing joystick data.
-						if (joy[joystick].stick[stick].flags & JOYFLAG_UNSIGNED) {
-							if (std::abs(joystickAxis->pos - 128) > 0 && std::fabs(joystickAxis->pos - 128) / 128 < deadZone) {
-								joystickAxis->pos = 128;
-								joystickAxis->d1 = joystickAxis->d2 = 0;
+					if (inputElements->at(elementsToCheck[i + 1]).JoyDirMapped()) {
+						axisUp = inputElements->at(elementsToCheck[i + 1]).GetAxis();
+						aimValues.m_Y = AnalogAxisValue(joystickIndex, axisUp);
+						isAxisMapped |= axisUp == axis;
+					}
+					if (!isAxisMapped) {
+						continue;
+					}
+
+					if (deadZoneType == DeadZoneType::CIRCLE) {
+						if (aimValues.MagnitudeIsLessThan(deadZone)) {
+							if (axisLeft != SDL_CONTROLLER_AXIS_INVALID) {
+								s_ChangedJoystickStates[joystickIndex].m_Axis[axisLeft] = Sign(axisLeft == axis ? -prevAxisValue : -device->m_Axis[axisLeft]);
+								s_ChangedJoystickStates[joystickIndex].m_DigitalAxis[axisLeft] = Sign(axisLeft == axis ? -prevDigitalValue : -device->m_DigitalAxis[axisLeft]);
+								device->m_Axis[axisLeft] = 0;
+								device->m_DigitalAxis[axisLeft] = 0;
 							}
-						} else {
-							if (std::abs(joystickAxis->pos) > 0 && std::fabs(joystickAxis->pos) / 128 < deadZone) { joystickAxis->pos = joystickAxis->d1 = joystickAxis->d2 = 0; }
+							if (axisUp != SDL_CONTROLLER_AXIS_INVALID) {
+								s_ChangedJoystickStates[joystickIndex].m_Axis[axisUp] = Sign(axisUp == axis ? -prevAxisValue : -device->m_Axis[axisUp]);
+								s_ChangedJoystickStates[joystickIndex].m_DigitalAxis[axisUp] = Sign(axisLeft == axis ? -prevDigitalValue : -device->m_DigitalAxis[axisUp]);
+								device->m_Axis[axisUp] = 0;
+								device->m_DigitalAxis[axisUp] = 0;
+							}
+						}
+					} else if (deadZoneType == DeadZoneType::SQUARE && deadZone > 0.0F) {
+						if (std::abs(value / 32767.0) < deadZone) {
```

Comment:
What is 32767? Should probably be pulled into something named rather than being magic number

Also, this is int / double, something should to be static_cast'd to specify math properly